crepecake/src/test/java/com/google/cloud/tools/crepecake/tar/TarStreamBuilderTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.tar;
+
+import com.google.cloud.tools.crepecake.blob.BlobStream;
+import com.google.common.io.CharStreams;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.util.function.Function;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.CompressorException;
+import org.apache.commons.compress.compressors.CompressorStreamFactory;
+import org.junit.Assert;
+import org.junit.Test;
+
+/** Tests for {@link TarStreamBuilder}. */
+public class TarStreamBuilderTest {
+
+  @Test
+  public void testBuild_compressed() throws IOException, URISyntaxException, CompressorException {
+    testBuild(
+        tarStreamBuilder -> {
+          try {
+            return tarStreamBuilder.toBlobStreamCompressed();
+          } catch (IOException ex) {
+            throw new RuntimeException(ex);
```

Comment:
shouldn't need to wrap these in runtime exceptions. If that's required for lambdas, then lambdas are inappropriate here.